source/jormungandr/jormungandr/scenarios/new_default.py

```diff
@@ -731,7 +731,10 @@ def merge_responses(responses):
         initial_feed_publishers = {}
         for fp in merged_response.feed_publishers:
             initial_feed_publishers[fp.id] = fp
-        merged_response.feed_publishers.extend((fp for fp in r.feed_publishers if fp.id not in initial_feed_publishers))
+
+        merged_response.feed_publishers.extend(fp for fp in r.feed_publishers
+                                               if fp.id not in initial_feed_publishers
+                                               and (debug or all('to_delete' not in j.tags for j in r.journeys)))
```

Comment:
If a response has multiple journeys, some deleted, some not, I don't think we will get what we want if journeys of a response have different feed_publisher.
I don't think this case happen for walking, bike and car. For BSS it could happen as a bss request can return a walking solution, but since we do not use any external system for bss it doesn't really matter.

Maybe a comment to explain that it can happen is enough as we do not have the problem right now and maybe never will.

Reply:
Then it may be a design flaw. The feed publisher should be linked to one journey if we really want to prevent this case.